writer.go

```diff
@@ -100,23 +100,36 @@ func NewGenericWriter[T any](output io.Writer, options ...WriterOption) *Generic
 	schema := config.Schema
 	t := typeOf[T]()
 
+	var genWriteErr error
 	if schema == nil && t != nil {
 		schema = schemaOf(dereference(t))
+		if len(schema.Columns()) == 0 {
+			genWriteErr = fmt.Errorf("cannot write %v: it has no columns (maybe it has no exported fields)", t)
+		}
 		config.Schema = schema
+	} else if schema != nil && len(schema.Columns()) == 0 {
+		genWriteErr = fmt.Errorf("cannot write %v: schema has no columns", t)
```

Comment:
Maybe this is not necessary... It will (possibly confusingly) produce zero rows, But nowhere else do we complain about the schema having zero rows (like in the lower-level `Writer` type). Presumably,  if the schema was explicitly set to zero columns... they won't actually attempt to write any rows? :shrug: